core/src/main/java/org/elasticsearch/gateway/DanglingIndicesState.java

```diff
@@ -119,12 +120,31 @@ void findNewAndAddDanglingIndices(MetaData metaData) {
                 try {
                     IndexMetaData indexMetaData = metaStateService.loadIndexState(indexName);
                     if (indexMetaData != null) {
-                        logger.info("[{}] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state", indexName);
-                        if (!indexMetaData.getIndex().getName().equals(indexName)) {
-                            logger.info("dangled index directory name is [{}], state name is [{}], renaming to directory name", indexName, indexMetaData.getIndex());
-                            indexMetaData = IndexMetaData.builder(indexMetaData).index(indexName).build();
+                        final String clusterUUID = metaData.clusterUUID();
+                        if (indexMetaData.isSameClusterUUID(clusterUUID)) {
+                            // the cluster metadata's cluster UUID is the same as the cluster UUID that created this index,
+                            // so the index shouldn't be imported as dangling but rather deleted and cleaned up, because the
+                            // dangling files are probably due to remnants from deleted indices (e.g. the node was offline while
+                            // the index was deleted) or failed cleanup
+                            logger.info("[{}] dangling index exists on the local file system, but it is deleted in the metadata of the " +
+                                        "same cluster that created it, deleting the index on the local file system", indexName);
+                            nodeEnv.deleteIndexDirectoryUnderLock(indexMetaData.getIndex(), new IndexSettings(indexMetaData, settings));
```

Comment:
I think we should make this an explicit handling in the dangling pipeline. Just import them into the map here and do the following:

```
    public void processDanglingIndices(MetaData metaData) {
        if (nodeEnv.hasNodeFile() == false) {
            return;
        }
        cleanupAllocatedDangledIndices(metaData);
        findNewAndAddDanglingIndices(metaData);
        cleanDeletedIndices(metaData); <-- this is where this logic will live, before allocation.
        allocateDanglingIndices();
    }
```

This will make it easier to test as well...